lib/logstash/outputs/http.rb

```diff
@@ -100,7 +102,12 @@ def receive(event)
     headers = event_headers(event)
 
     # Create an async request
-    request = client.send(@http_method, url, body: body, headers: headers, async: true)
+    request = client.send(@http_method, url, :body => body, :headers => headers, :async => true)
+
+    # with Maticore version < 0.5 using :async => true places the requests in an @async_requests
+    # list which is used & cleaned by Client#execute! but we are not using it here and we must
+    # purge it manually to avoid leaking requests.
+    client.clear_pending
```

Comment:
if we're doing this, do we still need to return the token to the pool in https://github.com/colinsurprenant/logstash-output-http/blob/fix/memleak/lib/logstash/outputs/http.rb#L114-L117?